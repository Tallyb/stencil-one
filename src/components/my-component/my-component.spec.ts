import { newSpecPage } from '@stencil/core/testing';

import { MyComponent } from './my-component';

it('override default values from attribute', async () => {
  const { root } = await newSpecPage({
    components: [MyComponent],
    html: `<my-component some-prop="value"></my-component>`,
  });

  expect(root).toEqualHtml(`
    <my-component first="value">
      value
    </my-component>
  `);

  expect(root.someProp).toBe('value');
});
