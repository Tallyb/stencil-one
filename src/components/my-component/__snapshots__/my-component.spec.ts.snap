// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
HostElement {
  "childNodes": Array [],
  "namespaceURI": "http://www.w3.org/1999/xhtml",
  "nodeName": "MY-COMPONENT",
  "nodeType": 1,
  "nodeValue": null,
  "ownerDocument": MockDocument {
    "childNodes": Array [
      MockDocumentTypeNode {
        "childNodes": Array [],
        "namespaceURI": null,
        "nodeName": "!DOCTYPE",
        "nodeType": 10,
        "nodeValue": null,
        "ownerDocument": [Circular],
        "parentNode": [Circular],
      },
      MockElement {
        "childNodes": Array [
          MockElement {
            "childNodes": Array [
              MockElement {
                "childNodes": Array [
                  MockTextNode {
                    "childNodes": Array [],
                    "nodeName": "#text",
                    "nodeType": 3,
                    "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "namespaceURI": null,
                "nodeName": "STYLE",
                "nodeType": 1,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
            ],
            "namespaceURI": null,
            "nodeName": "HEAD",
            "nodeType": 1,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
          MockElement {
            "childNodes": Array [
              [Circular],
            ],
            "namespaceURI": null,
            "nodeName": "BODY",
            "nodeType": 1,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
        ],
        "namespaceURI": null,
        "nodeName": "HTML",
        "nodeType": 1,
        "nodeValue": null,
        "ownerDocument": [Circular],
        "parentNode": [Circular],
      },
    ],
    "cookie": "",
    "defaultView": [MockWindow],
    "nodeName": "#document",
    "nodeType": 9,
    "nodeValue": null,
    "ownerDocument": null,
    "parentNode": null,
    "referrer": "",
  },
  "parentNode": MockElement {
    "childNodes": Array [
      [Circular],
    ],
    "namespaceURI": null,
    "nodeName": "BODY",
    "nodeType": 1,
    "nodeValue": null,
    "ownerDocument": MockDocument {
      "childNodes": Array [
        MockDocumentTypeNode {
          "childNodes": Array [],
          "namespaceURI": null,
          "nodeName": "!DOCTYPE",
          "nodeType": 10,
          "nodeValue": null,
          "ownerDocument": [Circular],
          "parentNode": [Circular],
        },
        MockElement {
          "childNodes": Array [
            MockElement {
              "childNodes": Array [
                MockElement {
                  "childNodes": Array [
                    MockTextNode {
                      "childNodes": Array [],
                      "nodeName": "#text",
                      "nodeType": 3,
                      "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                      "ownerDocument": [Circular],
                      "parentNode": [Circular],
                    },
                  ],
                  "namespaceURI": null,
                  "nodeName": "STYLE",
                  "nodeType": 1,
                  "nodeValue": null,
                  "ownerDocument": [Circular],
                  "parentNode": [Circular],
                },
              ],
              "namespaceURI": null,
              "nodeName": "HEAD",
              "nodeType": 1,
              "nodeValue": null,
              "ownerDocument": [Circular],
              "parentNode": [Circular],
            },
            [Circular],
          ],
          "namespaceURI": null,
          "nodeName": "HTML",
          "nodeType": 1,
          "nodeValue": null,
          "ownerDocument": [Circular],
          "parentNode": [Circular],
        },
      ],
      "cookie": "",
      "defaultView": [MockWindow],
      "nodeName": "#document",
      "nodeType": 9,
      "nodeValue": null,
      "ownerDocument": null,
      "parentNode": null,
      "referrer": "",
    },
    "parentNode": MockElement {
      "childNodes": Array [
        MockElement {
          "childNodes": Array [
            MockElement {
              "childNodes": Array [
                MockTextNode {
                  "childNodes": Array [],
                  "nodeName": "#text",
                  "nodeType": 3,
                  "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                  "ownerDocument": MockDocument {
                    "childNodes": Array [
                      MockDocumentTypeNode {
                        "childNodes": Array [],
                        "namespaceURI": null,
                        "nodeName": "!DOCTYPE",
                        "nodeType": 10,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                      [Circular],
                    ],
                    "cookie": "",
                    "defaultView": [MockWindow],
                    "nodeName": "#document",
                    "nodeType": 9,
                    "nodeValue": null,
                    "ownerDocument": null,
                    "parentNode": null,
                    "referrer": "",
                  },
                  "parentNode": [Circular],
                },
              ],
              "namespaceURI": null,
              "nodeName": "STYLE",
              "nodeType": 1,
              "nodeValue": null,
              "ownerDocument": MockDocument {
                "childNodes": Array [
                  MockDocumentTypeNode {
                    "childNodes": Array [],
                    "namespaceURI": null,
                    "nodeName": "!DOCTYPE",
                    "nodeType": 10,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                  [Circular],
                ],
                "cookie": "",
                "defaultView": [MockWindow],
                "nodeName": "#document",
                "nodeType": 9,
                "nodeValue": null,
                "ownerDocument": null,
                "parentNode": null,
                "referrer": "",
              },
              "parentNode": [Circular],
            },
          ],
          "namespaceURI": null,
          "nodeName": "HEAD",
          "nodeType": 1,
          "nodeValue": null,
          "ownerDocument": MockDocument {
            "childNodes": Array [
              MockDocumentTypeNode {
                "childNodes": Array [],
                "namespaceURI": null,
                "nodeName": "!DOCTYPE",
                "nodeType": 10,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
              [Circular],
            ],
            "cookie": "",
            "defaultView": [MockWindow],
            "nodeName": "#document",
            "nodeType": 9,
            "nodeValue": null,
            "ownerDocument": null,
            "parentNode": null,
            "referrer": "",
          },
          "parentNode": [Circular],
        },
        [Circular],
      ],
      "namespaceURI": null,
      "nodeName": "HTML",
      "nodeType": 1,
      "nodeValue": null,
      "ownerDocument": MockDocument {
        "childNodes": Array [
          MockDocumentTypeNode {
            "childNodes": Array [],
            "namespaceURI": null,
            "nodeName": "!DOCTYPE",
            "nodeType": 10,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
          [Circular],
        ],
        "cookie": "",
        "defaultView": [MockWindow],
        "nodeName": "#document",
        "nodeType": 9,
        "nodeValue": null,
        "ownerDocument": null,
        "parentNode": null,
        "referrer": "",
      },
      "parentNode": MockDocument {
        "childNodes": Array [
          MockDocumentTypeNode {
            "childNodes": Array [],
            "namespaceURI": null,
            "nodeName": "!DOCTYPE",
            "nodeType": 10,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
          [Circular],
        ],
        "cookie": "",
        "defaultView": [MockWindow],
        "nodeName": "#document",
        "nodeType": 9,
        "nodeValue": null,
        "ownerDocument": null,
        "parentNode": null,
        "referrer": "",
      },
    },
  },
}
`;
